src/WebJobs.Script/Host/IWorkerFunctionMetadataProvider.cs

```diff
@@ -0,0 +1,17 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using System.Collections.Immutable;
+using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs.Script.Workers.Rpc;
+
+namespace Microsoft.Azure.WebJobs.Script
+{
+    public interface IWorkerFunctionMetadataProvider
```

Comment:
Same with the other comment - recommend making this and other types `internal` if possible.